operator/controller/src/main/java/io/apicurio/registry/operator/resource/app/AppNetworkPolicyResource.java

```diff
@@ -31,6 +38,30 @@ public AppNetworkPolicyResource() {
     @Override
     protected NetworkPolicy desired(ApicurioRegistry3 primary, Context<ApicurioRegistry3> context) {
         var networkPolicy = APP_NETWORK_POLICY_KEY.getFactory().apply(primary);
+
+        Optional.ofNullable(primary.getSpec())
+                .map(ApicurioRegistry3Spec::getApp)
+                .map(AppSpec::getTls)
+                .ifPresent(tls -> {
+
+                    var httpsPolicy = new io.fabric8.kubernetes.api.model.networking.v1.NetworkPolicyPortBuilder()
+                            .withPort(new IntOrStringBuilder().withValue(8443).build()).build();
+
+                    var httpPolicy = new io.fabric8.kubernetes.api.model.networking.v1.NetworkPolicyPortBuilder()
+                            .withPort(new IntOrStringBuilder().withValue(8080).build()).build();
+
+                    if (tls.getInsecureRequests() != null && !tls.getInsecureRequests().equals("enabled")) {
```

Comment:
This condition can be moved to `io.apicurio.registry.operator.feat.TLS` behind a well-named static method, since it's used in multiple places. The optional could be moved there as well, since `tls` var is only used in the condition.

Reply:
`tls.getInsecureRequests()` might be null here, it's checked in the `AppServiceResource` below.